src/protocol/basics/reshare.rs

```diff
@@ -280,6 +346,37 @@ mod tests {
             }
         }
 
+        /// Validates that reshare protocol actually generates new xor shares using PRSS.
+        #[tokio::test]
+        async fn generates_unique_xor_shares() {
+            let world = TestWorld::new().await;
+
+            for &target in Role::all() {
+                let secret = thread_rng().gen::<BitArray40>();
+                let shares = world
+                    .semi_honest(secret, |ctx, share: XorShare<BitArray40>| async move {
+                        let record_id = RecordId::from(0);
+                        let ctx = ctx.set_total_records(1);
+
+                        // run reshare protocol for all helpers except the one that does not know the input
+                        if ctx.role() == target {
+                            // test follows the reshare protocol
+                            ctx.prss().generate_bit_arrays(record_id).into()
+                        } else {
+                            share.reshare(ctx, record_id, target).await.unwrap()
+                        }
```

Comment:
Why can't you just run the reshare protocol for all? Doesn't the reshare protocol just do this same behaviour when this helper is the target?

Reply:
I tried to factor out the common part but I realized it'd add more code than reduce them. Namely, `generate_bit_arrays` and `generate_fields` need a new function that returns `SecretSharing`. The calls to these functions are at the deepest part of the loop, and given this is a test code, I figured I would just copy them.